core/trino-main/src/main/java/io/trino/cost/StageRuntimeInfoProvider.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.trino.cost;
+
+import com.google.common.collect.ImmutableMap;
+import io.trino.sql.planner.PlanFragment;
+import io.trino.sql.planner.plan.PlanFragmentId;
+import io.trino.sql.planner.plan.PlanNode;
+
+import java.util.Map;
+
+import static io.trino.execution.scheduler.faulttolerant.OutputStatsEstimator.OutputStatsEstimateResult;
+import static java.util.Objects.requireNonNull;
+
+public class StageRuntimeInfoProvider
```

Comment:
RuntimeStatsProvider? Do we expect to have more than stats here?

Reply:
Why do we have `Stage` prefix in this class name. It suggests it allows for getting information for a single stage. While it covers whole query tree, right?